ipaserver/install/server/upgrade.py

```diff
@@ -946,8 +946,7 @@ def named_root_key_include():
 
 
 def named_update_global_forwarder_policy():
-    bind = bindinstance.BindInstance()
-    if not bindinstance.named_conf_exists() or not bind.is_configured():
+    if not bindinstance.named_conf_exists():
```

Comment:
The function name ``named_conf_exists`` is confusing. It checks that ``/etc/named.conf`` exists and is configured for IPA. How about you pass in ``bind`` instance to all functions and use ``bind.is_configured()`` instead?

Reply:
`bind.is_configured()` is the link to `bindinstance.named_conf_exists()`. Perhaps, to speed up the checking of that condition the module scope function was added (no need to instantiate of `BindInstance`).
Each `named_` function in the upgrade code uses the same logic:
```
 542     if not bindinstance.named_conf_exists():                                    
 543         # DNS service may not be configured                                     
 544         logger.info('DNS is not configured')                                    
 545         return False
```
This results in each of these functions read named.conf. I'll move this check out of the scope of `named_*` and check only once.